var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -204,148 +310,157 @@ def url_for_version(self, version):
-        "compress",
-        "coreshell",
-        "dielectric",
-        "dipole",
-        "electrode",
-        "granular",
-        "intel",
-        "kspace",
-        "kokkos",
-        "latte",
-        "manybody",
-        "mc",
-        "meam",
-        "misc",
-        "mliap",
-        "ml-hdnnp",
-        "ml-iap",
-        "ml-snap",
-        "molecule",
-        "mpiio",
-        "opt",
-        "peri",
-        "plugin",
-        "poems",
-        "python",
-        "qeq",
-        "replica",
-        "rigid",
-        "shock",
-        "snap",
-        "spin",
-        "srd",
-        "voronoi",
-        "user-atc",
-        "user-adios",
-        "user-awpmd",
-        "user-bocs",
-        "user-brownian",
-        "user-cgsdk",
-        "user-colvars",
-        "user-diffraction",
-        "user-dpd",
-        "user-drude",
-        "user-eff",
-        "user-fep",
-        "user-h5md",
-        "user-hdnnp",
-        "user-intel",
-        "user-lb",
-        "user-manifold",
-        "user-meamc",
-        "user-mesodpd",
-        "user-mesont",
-        "user-mgpt",
-        "user-misc",
-        "user-mofff",
-        "user-molfile",
-        "user-netcdf",
-        "user-omp",
-        "user-phonon",
-        "user-plumed",
-        "user-ptm",
-        "user-qtb",
-        "user-reaction",
-        "user-reaxc",
-        "user-sdpd",
-        "user-smd",
-        "user-smtbq",
-        "user-sph",
-        "user-tally",
-        "user-uef",
-        "user-yaff",
-        "atc",
-        "adios",
-        "awpmd",
-        "bocs",
-        "brownian",
-        "cg-sdk",
-        "colvars",
-        "diffraction",
-        "dpd-basic",
-        "dpd-meso",
-        "dpd-react",
-        "dpd-smooth",
-        "drude",
-        "eff",
-        "extra-compute",
-        "extra-dump",
-        "extra-fix",
-        "extra-molecule",
-        "extra-pair",
-        "fep",
-        "h5md",
-        "interlayer",
-        "latboltz",
-        "machdyn",
-        "manifold",
-        "mesont",
-        "mgpt",
-        "mofff",
-        "molfile",
-        "netcdf",
-        "openmp-package",
-        "orient",
-        "phonon",
-        "plumed",
-        "ptm",
-        "qtb",
-        "reaction",
-        "reaxff",
-        "smtbq",
-        "sph",
-        "tally",
-        "uef",
-        "yaff",
-    ]
-
-    for pkg in supported_packages:
+    supported_packages = {
+        "asphere": {},
+        "body": {},
+        "bpm": {"when": "@20220504:"},
+        "class2": {},
+        "colloid": {},
+        "compress": {},
+        "coreshell": {},
+        "dielectric": {"when": "@20210702:"},
+        "dipole": {},
+        "electrode": {"when": "@20220504:"},
+        "granular": {},
+        "intel": {},
+        "kim": {},
+        "kspace": {"default": True},
+        "kokkos": {},
+        "latte": {},
+        "manybody": {"default": True},
+        "mc": {},
+        "meam": {"when": "@20210728:"},
```

Comment:
Meam existed before, in my tentative PR I had the following conflicts:
```python
    conflicts(
        '+meam', when='@20190104:20210527',
        msg='+meam was removed from version @20190104 to version @20210527, ' +                                                          
            'use +user-meamc instead')
    conflicts(
        '+user-meamc', when='@:20181212,20210702:',
        msg='+user-meamc only supported after version @20181212 and before ' +                                                           
            'version @20210702, use +meam instead')
```

Reply:
Went over all packages and corrected their version ranges. Wrote myself a little tool that went over the git history:
https://gist.github.com/rbberger/fdaa38ff08e5961c4741624a4719cdb6